### PR TITLE
Make the project rails 7.1 compatible

### LINKED ISSUE #249: incompatible with rails 7.1
The latest version of minitest-rails is pinned to `gem "rails", "~> 7.0.0"`, so it is unfortunately not usable with the Rails 7.1 release candidate out of the box.

I was able to fork minitest-rails and change to `gem "rails", "~> 7.1.0.rc1"`, and that seems to work.

Is there any reason why minitest-rails can't simply enforce a minimum version requirement (i.e. `gem "rails", ">= 7.0.0"`), rather than pinning to specific Rails versions? This is the only gem in our app, out of ~110 gems, that pins to a specific Rails version, making upgrading more painful than necessary.

I'm happy to submit a pull request for this change if it is welcome.

cc @blowmage

### PR DESCRIPTION
**Pull request by manuelvanrijn**
I've made the following changes to get the project running on a rails 7.1 project.
Aside from the changes, I've also included ruby 3.1 in the CI matrix: https://github.com/manuelvanrijn/minitest-rails/actions/runs/6459990864

Hope this helps to get the project in shape for this new rails version :)

Fixes #249

### CODE REVIEW

**Comment by skateman** [comment]
@blowmage is anything else needed to get this merged 🙏 ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,7 +56,7 @@ jobs:
 
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.7
+          ruby-version: 3.3
```
Comment (blowmage) [inline .github/workflows/ci.yml:59]: 3.3 isn't released yet

File: test/helper.rb [hunk 2]
```diff
@@ -21,7 +21,6 @@
 require "action_mailer"
 
 require "minitest-rails"
-require "rails/test_help"
```
Comment (blowmage) [inline test/helper.rb:24]: why was this removed?

**Comment by blowmage** [comment]
[minitest-rails 7.1.0](https://rubygems.org/gems/minitest-rails/versions/7.1.0) is released.

Thank you for this PR. I had to update some additional files and this PR ended up with some merge conflicts, which is why it wasn't merged. My apologies. I would have preferred to get you the commit, but I figured you probably wanted the release more. Please do contribute again in the future.
